### DISCUSSION TITLE
Tensor Mechanics Master Action Outputs not compatible with the new system.

### DISCUSSION THREAD

**Discussion by Edward-Eth** [idaholab/moose#22765] [Q&A Modules: Solid mechanics] [answered]
I recently discovered that some tensor mechanics generate outputs arguments in the master action block do not work when used with the new system for tensor mechanics.

For example, vonmises_stress does not work, as it is hard coded to calculate the vonmises value using the global tensor 'stress'. However, when using the new system, stresses are stored in the global tensor 'cauchy_stress'. This means to extract vonmises stresses you must use a rank two aux kernel. 

This is not a huge issue, but it would be good if the master action could intelligently switch between looking for stress and cauchy stress depending on the system in use.

**Comment by GiudGiud** [comment]
@bwspenc who else should we tag to get some feedback / opinions on this

**Reply by hugary1995** [reply]
This issue does exist and should be fixed.

**Reply by bwspenc** [reply]
@reverendbedford @dschwen

**Comment by bwspenc** [comment, marked as answer]
Yeah, this is an issue. We have long had a convention that the property named `stress` was the Cauchy stress. The developers of the new system were explicit in their naming of the stress because there were multiple types of stress computed. A few options:
1) Change the name of `cauchy_stress` to `stress` in the new system
2) Change the name of `stress` to `cauchy_stress` in the old system
3) Have both the new and old system compute properties called `cauchy_stress` and `stress`

I think option 1 would be the easiest, but maybe that's not the best route, because there's certainly something to be said for being explicit in the naming. I think this is my favorite, though.

Option 2 would cause huge amounts of breakage. Also, we have to think about how far to carry the convention of explicitly spelling out the stress type. Would we call it `vonmises_cauchy_stress`, for instance? 

Option 3 would require extra compute resources and would potentially be confusing.

I'm interested in what others think.

**Reply by reverendbedford** [reply]
I think it has to be option 1, unless we can think of something else entirely.

**Reply by GiudGiud** [reply]
Rest of MOOSE is aiming to be more explicit and less vague in their names. Having things named vaguely or imprecisely hurts the next generation of users and developers as they have to learn the imprecision, often through a mistake and its consequences

**Reply by bwspenc** [reply]
@GiudGiud I totally appreciate the desire to define things precisely. It's always a challenge finding the right balance between that and usability. I'm not sure what the right answer is here. We need to think through the full consequences of whatever path we choose.

**Reply by hugary1995** [reply]
How about defining a new parameter in the TM action called "output_stress_measure" or something like that which defaults to "stress" if `new_system = false`, and to "cauchy_stress" if `new_system = true`.

**Reply by bwspenc** [reply]
@hugary1995 I like that idea -- something along the lines of that could be a good solution.

**Reply by hugary1995** [reply]
Okay, that sounds like the end of the bike shedding time. I can handle this when I get a chance if no one beats me to it. #22776

**Reply by Edward-Eth** [reply]
Marking this as solved pending a fix being implemented, thanks for the quick response 👍
